persist/fs/commitlog/reader.go

```diff
@@ -375,148 +375,42 @@ func (r *reader) decodeAndHandleMetadata(
 			Namespace:   ident.BinaryID(namespace),
 			Shard:       decoded.Shard,
 		}
-		r.metadata.lookup[entry.Index] = metadata
-		pendingMetadata, ok := r.metadata.pending[entry.Index]
+		metadataLookup[entry.Index] = seriesMetadata{
```

Comment:
reminder note for us to talk about the lifecycle of the IDs allocated here:
- do you transfer ownership of allocated IDs to the returned struct
- what about series which only have a metadata entry (for w/e reason)

Reply:
cool lets sync in person

Reply:
per offline chat, lets open the issue for transfer of ownership to shard